doc/pbh/pbh-design.md

```diff
@@ -0,0 +1,758 @@
+```bash
+root@sonic:/home/admin# show pbh table
+Name       Interface        Description
+---------  ---------------  ---------------
+pbh_table  Ethernet0        NVGRE and VxLAN
+           Ethernet4
+           PortChannel0001
+           PortChannel0002
+```
+
+**The following command shows rule configuration:**
+```bash
+root@sonic:/home/admin# show pbh rule
+Table      Rule    Priority    Match                            Hash               Action
+---------  ------  ----------  -------------------------------  -----------------  -------------
+pbh_table  nvgre   1           GRE_KEY: 0x2500/0xffffff00       inner_ip_proto     SET_ECMP_HASH
+                               INNER_ETHER_TYPE: 0x86dd/0xffff  inner_l4_dst_port
+                                                                inner_l4_src_port
+                                                                inner_dst_ipv6
+                                                                inner_src_ipv6
+           vxlan   2           IP_PROTOCOL: 0x11/0xff           inner_ip_proto     SET_LAG_HASH
+                               L4_DST_PORT: 0x12b5/0xffff       inner_l4_dst_port
+                               INNER_ETHER_TYPE: 0x0800/0xffff  inner_l4_src_port
+                                                                inner_dst_ipv4
+                                                                inner_src_ipv4
+```
+
+**The following command shows hash configuration:**
+```bash
+root@sonic:/home/admin# show pbh hash
+Name               Field              Mask       Sequence    Symmetric
+-----------------  -----------------  ---------  ----------  -----------
+inner_ip_proto     INNER_IP_PROTOCOL             1           No
+inner_l4_dst_port  INNER_L4_DST_PORT             2           Yes
+inner_l4_src_port  INNER_L4_SRC_PORT             2           Yes
+inner_dst_ipv4     INNER_DST_IPV4     255.0.0.0  3           Yes
+inner_src_ipv4     INNER_SRC_IPV4     0.0.0.255  3           Yes
+inner_dst_ipv6     INNER_DST_IPV6     FFFF::     4           Yes
+inner_src_ipv6     INNER_SRC_IPV6     ::FFFF     4           Yes
+```
+
+**The following command shows statistics:**
+```bash
+root@sonic:/home/admin# show pbh statistics
+Table      Rule    Packets Count    Bytes Count
+---------  ------  ---------------  -------------
+pbh_table  nvgre   0                0
+           vxlan   0                0
+```
+
+## 2.7 DPB YANG model
+
+A new YANG model `sonic-pbh.yang` will be added to `sonic-buildimage/src/sonic-yang-models/yang-models`  
+in order to provide support for DPB and management framework.
+
+**Skeleton code:**
+```yang
+module sonic-pbh {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/Azure/sonic-pbh";
+    prefix pbh;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    import sonic-types {
+        prefix stypes;
+        revision-date 2019-07-01;
+    }
+
+    import sonic-extension {
+        prefix ext;
+        revision-date 2019-07-01;
```

Comment:
We dont need revision-date any more, please remove it.

Reply:
@venkatmahalingam we still use `revision-date` in latest master, so can't remove it so far.
Also, according to RFC 7950 it's a bad idea to completely remove/abandon it:

https://tools.ietf.org/html/rfc7950#section-7.1.5
```
When the optional "revision-date" substatement is present, any
typedef, grouping, extension, feature, and identity referenced by
definitions in the local module are taken from the specified revision
of the imported module.  It is an error if the specified revision of
the imported module does not exist.  If no "revision-date"
substatement is present, it is undefined from which revision of the
module they are taken.
```

Reply:
There is an effort in progress to remove the revision date, please refer to the new YANG models in review for better clarity.

Reply:
@venkatmahalingam done